magento/ece-tools
# MCLOUD-6375: MC-34668: Introduce environment variable to store a list of quality p…

https://jira.corp.magento.com/browse/MCLOUD-6375: 

In scope of the story [Magento Quality Patches](https://jira.corp.magento.com/browse/MC-34580) added environment variable QUALITY_PATCHES. By default, it's empty and doesn't affect the deployment process. We need it in develop branch to proceed with [adopting CI builds](https://jira.corp.magento.com/browse/MC-34666) for Magento Quality Patches 

### Description
- Added environment variable QUALITY_PATCHES
- Updated patch manager to use QUALITY_PATCHES variable
- Implemented copying patch.log file from build phase to {magento_root}/var/log directory

### Testing scenario
- Check if your GitHub token has access to https://github.com/magento-mpi/magento-cloud-patches.git, https://github.com/magento/quality-patches.git, https://github.com/magento-mpi/ece-tools.git
- Add to composer.json 

```
 "repositories": {
        "repo": {
            "type": "composer",
            "url": "https://repo.magento.com"
        },
        "mcp": {
            "type": "vcs",
            "url": "https://github.com/magento-mpi/magento-cloud-patches.git"
        },
        "mqp": {
            "type": "vcs",
            "url": "https://github.com/magento/quality-patches.git"
        },
        "ece": {
            "type": "vcs",
            "url": "https://github.com/magento-mpi/ece-tools.git"
        }
    },
    "require": {
        "magento/magento-cloud-patches": "dev-MC-34580 as 1.0.99",
        "magento/quality-patches": "dev-master as 1.0.999",
        "magento/ece-tools": "dev-MC-34668 as 2002.1.99"
    },
```

- Add to .magento.env.yaml
```
 stage:
    build:
        QUALITY_PATCHES:
            - MCTEST-1001
```

- Install Magento
- Check /var/log/patch.log file contains 
```
INFO: Patch MCTEST-1001 has been applied {"file":"/app/vendor/magento/quality-patches/src/../patches/MCTEST-1001__TEST_changes__2.3.5_ce.patch"}
INFO: Patch MCTEST-1001 has been applied {"file":"/app/vendor/magento/quality-patches/src/../patches/MCTEST-1001__TEST_changes__2.3.5_ee.patch"}
```
- Verify that patches https://github.com/magento/quality-patches/blob/master/patches/MCTEST-1001__TEST_changes__2.3.5_ce.patch and https://github.com/magento/quality-patches/blob/master/patches/MCTEST-1001__TEST_changes__2.3.5_ee.patch were applied (check sources in vendor folder) 

### Release notes

Added the `QUALITY_PATCHES` environment variable to specify one or more patches to apply during the deployment process. Currently, this variable is for Magento internal use only.

### Associated documentation updates



### Contribution checklist
